RecoLocalCalo/EcalRecProducers/plugins/EcalESProducerGPU.h

```diff
@@ -0,0 +1,45 @@
+#ifndef RecoLocalCalo_EcalRecProducers_src_EcalESProducerGPU_h
+#define RecoLocalCalo_EcalRecProducers_src_EcalESProducerGPU_h
+
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Utilities/interface/typelookup.h"
+#include "FWCore/Framework/interface/eventsetuprecord_registration_macro.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include <iostream>
+
+template<typename Target, typename Source, typename Record>
+class EcalESProducerGPU : public edm::ESProducer {
+public:
+    explicit EcalESProducerGPU(edm::ParameterSet const& ps) 
+        : label_{ps.getParameter<std::string>("label")}
+    {
+        std::string name = ps.getParameter<std::string>("ComponentName");
+        setWhatProduced(this, name);
+    }
+   
+    std::unique_ptr<Target> produce(Record const& record) {
+        // retrieve conditions in old format 
+        edm::ESTransientHandle<Source> product;
+        record.get(label_, product);
+
+        return std::make_unique<Target>(*product);
```

Comment:
Is it guaranteed that the `Target` will not store a pointer/reference to `product` or anything contained by `product`?